### PR TITLE
Slight simplification of accessibleRepositoryCondition

### PR DESCRIPTION
**Pull request by zeripath**
Very slight simplification of accessibleRepositoryCondition - remove one of the In queries.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by lunny** [comment]
It looks good, but why CI failed.

**Comment by zeripath** [comment]
I suspect it's unrelated and may be related to the start of summertime in many places?

File: models/repo_list.go [hunk 1]
```diff
@@ -353,11 +353,13 @@ func accessibleRepositoryCondition(user *User) builder.Cond {
 		// 1. Be able to see all non-private repositories that either:
 		cond = cond.Or(builder.And(
 			builder.Eq{"`repository`.is_private": false},
-			builder.Or(
-				//   A. Aren't in organisations  __OR__
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.Eq{"type": UserTypeOrganization})),
-				//   B. Isn't a private organisation. Limited is OK as long as we're logged in.
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.In("visibility", orgVisibilityLimit))))))
+			builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(
+				builder.Or(
```
Comment (zeripath) [inline models/repo_list.go]: oops de morgan's law...

```suggestion
				builder.And(
```

File: models/repo_list.go [hunk 2]
```diff
@@ -353,11 +353,13 @@ func accessibleRepositoryCondition(user *User) builder.Cond {
...
 		cond = cond.Or(builder.And(
 			builder.Eq{"`repository`.is_private": false},
-			builder.Or(
-				//   A. Aren't in organisations  __OR__
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.Eq{"type": UserTypeOrganization})),
-				//   B. Isn't a private organisation. Limited is OK as long as we're logged in.
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.In("visibility", orgVisibilityLimit))))))
+			builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(
+				builder.And(
+					//   A. Aren't in organisations  __OR__
```
Comment (zeripath) [inline models/repo_list.go]:
```suggestion
```

File: models/repo_list.go [hunk 3]
```diff
@@ -353,11 +353,13 @@ func accessibleRepositoryCondition(user *User) builder.Cond {
...
+					builder.Eq{"type": UserTypeOrganization},
+					//   B. Isn't a private organisation. Limited is OK as long as we're logged in.
```
Comment (zeripath) [inline models/repo_list.go]:
```suggestion
```

File: models/repo_list.go [hunk 4]
```diff
@@ -353,11 +353,13 @@ func accessibleRepositoryCondition(user *User) builder.Cond {
 		// 1. Be able to see all non-private repositories that either:
 		cond = cond.Or(builder.And(
 			builder.Eq{"`repository`.is_private": false},
-			builder.Or(
-				//   A. Aren't in organisations  __OR__
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.Eq{"type": UserTypeOrganization})),
-				//   B. Isn't a private organisation. Limited is OK as long as we're logged in.
-				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.In("visibility", orgVisibilityLimit))))))
+			builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(
```
Comment (zeripath) [inline models/repo_list.go:357]:
```suggestion
			// 2. Aren't in an private organisation or limited organisation if we're not logged in
			builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(
```

**Review by lafriks — APPROVED** [review]

**Review by mrsdizzie — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**